### PR TITLE
catch2_3 3.3.2 -> 3.4.0

### PR DESCRIPTION
**Pull request by theShmoo**
## Description of changes

catch2_3 from version 3.3.2 to version 3.4.0
[Release Notes of Catch2 3.4.0](https://github.com/catchorg/Catch2/releases/tag/v3.4.0)

### CODE REVIEW

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 248096` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>catch2_3</li>
    <li>jarowinkler-cpp</li>
    <li>python310Packages.jarowinkler</li>
    <li>python310Packages.jarowinkler.dist</li>
    <li>python311Packages.jarowinkler</li>
    <li>python311Packages.jarowinkler.dist</li>
    <li>sonic-pi</li>
    <li>sv-lang</li>
    <li>waybar</li>
    <li>waybar-hyprland</li>
    <li>xpano</li>
    <li>yuzu-early-access</li>
    <li>yuzu-mainline</li>
  </ul>
</details>

File: pkgs/development/libraries/catch2/3.nix [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
   pname = "catch2";
-  version = "3.3.2";
+  version = "3.4.0";
 
   src = fetchFromGitHub {
     owner = "catchorg";
     repo = "Catch2";
     rev = "v${version}";
-    hash = "sha256-t/4iCrzPeDZNNlgibVqx5rhe+d3lXwm1GmBMDDId0VQ=";
+    hash = "sha256-EikouBS3VxcxbHGvab0rQzh2Q7oHam7BbniCv7LfrLs=";
```
Comment (nim65s) [inline pkgs/development/libraries/catch2/3.nix:16]: I don't have the same hash:
```
error: hash mismatch in fixed-output derivation '/nix/store/kzna5978hkkm0ll7zrazkmw97cmpf08z-source.drv':
         specified: sha256-EikouBS3VxcxbHGvab0rQzh2Q7oHam7BbniCv7LfrLs=
            got:    sha256-DqGGfNjKPW9HFJrX9arFHyNYjB61uoL6NabZatTWrr0=

**Review by nim65s — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/catch2/3.nix (hunk 1 above)
Comment (theShmoo) [reply pkgs/development/libraries/catch2/3.nix]: Thanks for checking!

Weird. I created the hash with the following commands:

`nix-prefetch-url --type sha256 https://github.com/catchorg/Catch2/archive/refs/tags/v3.4.0.tar.gz`

which returns

`1fxcvyrbz0kqdv0nwsh7p91pcf235fynkbvidhqifmxp2jw2ha8j`

which i converted with

`nix hash to-sri --type sha256 1fxcvyrbz0kqdv0nwsh7p91pcf235fynkbvidhqifmxp2jw2ha8j`

to

`sha256-EikouBS3VxcxbHGvab0rQzh2Q7oHam7BbniCv7LfrLs=`

but i fixed the commit with your hash.

Comment (nim65s) [reply pkgs/development/libraries/catch2/3.nix]: I have no idea how we should compute them. Usually, I'll just increment a random number inside the old, and start the build. 
build tools will then fail and give me the right one :)

Comment (theShmoo) [reply pkgs/development/libraries/catch2/3.nix]: Thank you! :)

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 248096` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>catch2_3</li>
    <li>jarowinkler-cpp</li>
    <li>python310Packages.jarowinkler</li>
    <li>python310Packages.jarowinkler.dist</li>
    <li>python311Packages.jarowinkler</li>
    <li>python311Packages.jarowinkler.dist</li>
    <li>sonic-pi</li>
    <li>sv-lang</li>
    <li>waybar</li>
    <li>waybar-hyprland</li>
    <li>xpano</li>
    <li>yuzu-early-access</li>
    <li>yuzu-mainline</li>
  </ul>
</details>

**Review by nim65s — APPROVED** [review]

**Comment by dotlambda** [comment]
Duplicate of https://github.com/NixOS/nixpkgs/pull/243485

File: pkgs/development/libraries/catch2/3.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/libraries/catch2/3.nix]: You have to use `nix-prefetch-url --unpack`.

**Pull request closed without merging**